src/disjoint_set.jl

```diff
@@ -111,13 +115,16 @@ function root_union!(s::IntDisjointSets, x::Integer, y::Integer)
 end
 
 """
-    push!(s::IntDisjointSets)
+    push!(s::IntDisjointSets{T})
 
 Make a new subset with an automatically chosen new element x.
-Returns the new element.
+Returns the new element. Throw an `ArgumentError` if the
+capacity of the set would be exceeded.
 """
-function push!(s::IntDisjointSets)
-    x = length(s) + 1
+function push!(s::IntDisjointSets{T}) where {T<:Integer}
+    l = length(s)
+    l < typemax(T) || throw(ArgumentError(_intdisjointsets_bounds_err_msg(T))) 
+    x = l + 1
     push!(s.parents, x)
     push!(s.ranks, 0)
```

Comment:
```suggestion
    push!(s.ranks, zero(T))
```

Reply:
similar changes for 1, required in the rest of the code. eg in L110, 113

Reply:
absolutely.